Automattic/jetpack
# Enables Jetpack without the need of an authorized user

This PR makes Jetpack active as soon as there is a site level connection, not requiring a user connection in order to work. (aka what we have been calling User-Less).

#### Changes proposed in this Pull Request:

* Enables Jetpack without the need of an authorized user
* Adjusts the way we pass the `userless` GET-parameter to WP.com

#### Jetpack product discussion


p1HpG7-aJm-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






Test the connection flow in every possible way:

* Connect both from the Jetpack Dashboard and from the main wp-admin dashboard
* Only site level connection via in-place connection
* Only site level connection without 3rd party cookies support (NOT POSSIBLE YET, REQUIRES PATCH TO WPCOM)
* Reconnection after a broken connection (both when only connected at a site level and with a full connection)
* Full connection via  in-place connection and alternative flows (use `define( 'JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME', true );`

Also

* Go to Tools > Site Health
* Make sure tests regarding the connection owner are skipped if there's no user connection
![Captura de Tela 2021-04-22 às 17 57 24](https://user-images.githubusercontent.com/971483/115787313-616fb780-a398-11eb-8da1-80a9bbe967ef.png)
* Make sure these tests are NOT skipped if there's a connection owner (even with broken tokens)
